src/stores/FilterBarStore.js

```diff
@@ -0,0 +1,130 @@
+var SharedUtils = require('../utils/SharedUtils');
+
+export class FilterBarStore {
+  constructor(configuration) {
+    this.CHANGE_EVENT = 'change';
+    this.eventEmitter = new EventEmitter();
+
+    this.id = configuration.id;
+    this.persistent = configuration.persistent;
+    this.url = configuration.searchUrl;
+    this.searchUrl = configuration.searchUrl;
+    this.saveSearchUrl = configuration.saveSearchUrl;
+    this.savedSearchUrl = configuration.savedSearchUrl;
+    this.exportResultsUrl = configuration.exportResultsUrl;
+    this.filters = configuration.filters;
+
+    var filter, filterUid;
+
+    for (var i = 0; i < Object.keys(this.filters).length; i++) {
+      filterUid = Object.keys(this.filters)[i];
+      filter = this.filters[filterUid];
+
+      if (filter.url) {
+        SharedUtils.ajaxGet(filter.url, 'json', function(filterUid) { return function(response) { this.updateFilter(filterUid, 'options', response) } }(filterUid).bind(this));
+      }
+    }
+
+    this.receieveSavedSearches();
+  }
+
+  getId() {
+    return this.id;
+  }
+
+  getSearchUrl() {
+    return this.searchUrl;
+  }
+
+  getSaveSearchUrl() {
+    return this.saveSearchUrl;
+  }
+
+  getSavedSearches() {
+    return this.savedSearches;
+  }
+
+  getSavedSearch(searchId) {
+    return this.savedSearches[searchId];
+  }
+
+  getFilter(filterUid) {
+    return this.filters[filterUid]
+  }
+
+  getDisabled() {
+    var disabledFilters = {};
+    for (var filterUid in this.filters) {
+      if (this.filters[filterUid].enabled === false) {
+        disabledFilters[filterUid] = this.filters[filterUid];
+      }
+    }
+    return disabledFilters;
+  }
+
+  getEnabled() {
+    var enabledFilters = {};
+    for (var filterUid in this.filters) {
+      if (this.filters[filterUid].enabled === true) {
+        enabledFilters[filterUid] = this.filters[filterUid];
+      }
+    }
+    return enabledFilters;
+  }
+
+  getQuery() {
+    var enabledFilters = Object.keys(this.getEnabled()).map(function(filterUid) {
+      var filter = this.getFilter(filterUid);
+      return {
+        uid: filterUid,
+        type: filter.type,
+        field: filter.field,
+        value: filter.value
+      }
+    },this);
+    return enabledFilters.length > 0 ? JSON.stringify(enabledFilters) : '';
+  }
+
+  /* Mutation Methods */
+  receieveSavedSearches() {
```

Comment:
This definitely doesn't live here.

FilterBarStore seems to be doing too much. I can see 2 clear class you can pull this into
1. FilterStore:  A collection of filters and some state for each one .e.g Is each filter disabled
2. SavedSearchStore: A collection of saved searches

Once again we can pull all the ajax out into a client and then use it in an action